plugin/pkg/admission/unsafedeletepolicy/admission.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unsafedeletepolicy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/admission"
+	genericadmissioninit "k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+)
+
+// PluginName is the name of the admission plugin
+const PluginName = "AllowUnsafeMalformedObjectDeletion"
+
+var (
+	prefix           = strings.ToLower(PluginName) + ".admission.k8s.io/"
+	auditDecisionKey = prefix + "decision"
+	auditReasonKey   = prefix + "reason"
+)
```

Comment:
nit: just turn these into two `const`s.  It makes it easier to find via simple string searches.